# Informations

- [Nanoseq](https://github.com/nf-core/nanoseq) is a bioinformatics analysis pipeline that can be used to perform basecalling, demultiplexing, mapping and QC of Nanopore DNA/RNA sequencing data.
- At the moment (Oct 2020), it presents a problem (see [issue](https://github.com/nf-core/nanoseq/issues/77)), and has to be used in two steps.
- The singularity images that working are stored in `/groups/dog/matthias/singularity`.

# Usage

1. Specify in the `input_second_step.csv` file the path to the .fatsq.gz that will be generated by guppy (`/results/guppy/fastq/<sample_name>.fatsq.gz`). Example : If the sample name in your `input_first_step.csv` is CML10, write `./results/guppy/fastq/CML10.fastq.gz` in the 2nd column of `input_second_step.csv`. Example of `--input file`are shown [here](https://github.com/nf-core/nanoseq/blob/master/docs/usage.md#--input). 
2. Specify the location of your data in the sbatch_launcher.sh
3. Submit your job to SLURM.

# Guppy mode

To print all `genouest` guppy workflows :
```
/groups/dog/script/ont/ont-guppy/bin/guppy_basecaller --print_workflows
```
